### PR TITLE
Make most of the Vuex store modules namespaced

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #205 by @dhruv

## Description

This PR moves most of the Vuex store modules from `store-modules` to `store` directory. This way, Nuxt creates namespaced modules out of them, which are easier to maintain.

## Technical details

To make the transition to namespaced modules more atomic, one large #229 is replaced with #242 and this PR. Also, the `filter-store` and `search-store` are left as is because they are tightly coupled, and will be refactored in a later PR.
There are two things that needed to be changed in every store module:
- namespaced modules require state to be a function that returns the state object, instead of an object
- namespaced modules require actions to be plain object, so to pass data fetcher services into actions, `createAction` functions were added. This makes it possible to switch services with mocks when testing.

## Tests

Run the tests, and run the app to verify that no error messages appear in the console.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
Excellent. Tests pass and everything is working. I went through a little manual integration test suite haha, and everything works as it should. What a large amount of work!

File: src/pages/feedback.vue [hunk 1]
```diff
@@ -97,16 +97,12 @@ export const FeedbackPage = {
...
-    ...mapState(['isEmbedded']),
-    isReportingBug() {
-      return this.$store.state.isReportingBug
-    },
-    bugReported() {
-      return this.$store.state.bugReported
-    },
-    bugReportFailed() {
-      return this.$store.state.bugReportFailed
-    },
```
Comment (sarayourfriend) [inline src/pages/feedback.vue:109]: Just a curiosity, but why were these not using `mapState` before?

**Review by sarayourfriend — COMMENTED** [review]
What's the difference between this PR and #229?

**Comment by sarayourfriend** [comment]
> What's the difference between this PR and #229?

Oh sorry, I should have read the description more closely. Never mind!

File: src/pages/feedback.vue (hunk 1 above)
Comment (obulat) [reply src/pages/feedback.vue:109]: I am not sure, probably because that was the understanding at the time of writing of this code.

**Review by sarayourfriend — APPROVED** [review]
Looks good to me! Code changes make sense and it tested well locally. Nice refactor!

**Pull request merged**